app/models/quiz_item.rb

```diff
@@ -0,0 +1,30 @@
+require 'json'
+
+class QuizItem < Item
+  has_many :quiz_question_choices, class_name: 'QuizQuestionChoice', foreign_key: 'question_id', inverse_of: false, dependent: :nullify
+
+  def edit
+  end
+
+  def view_question_text
+    choices = quiz_question_choices.map do |choice|
+      {
+        text: choice.txt,
+        is_correct: choice.iscorrect?
+      }
+    end
+
+    {
+      question_text: txt,
+      question_type: type,
+      question_weight: weight,
+      choices: choices
+    }.to_json
+  end
+
+  def complete
+  end
+
+  def view_completed_question(user_answer = nil)
+  end
+end
```

Comment:
Please add carriage returns after the final line of files to suppress this ugly warning.